eosmetrics/emtr-event-recorder.c

```diff
@@ -1277,4 +1278,46 @@ emtr_event_recorder_record_stop_sync (EmtrEventRecorder *self,
                TRUE /* is_synchronous */);
 }
 
+/**
+ * emtr_event_recorder_start_aggregate_timer:
+ * @self: an #EmtrEventRecorder
+ * @event_id: an RFC 4122 UUID representing the type of event that took place
+ * @aggregate_key: the identifier used to aggregate data
+ * @auxiliary_payload: miscellaneous data to associate with the events. Must
+ * not contain maybe variants as they are not compatible with D-Bus.
+ *
+ * Requests the metrics daemon to create an aggregate timer.
+ */
+EmtrAggregateTimer *
+emtr_event_recorder_start_aggregate_timer (EmtrEventRecorder *self,
+                                           const gchar       *event_id,
+                                           GVariant          *aggregate_key,
+                                           GVariant          *auxiliary_payload)
+{
+  EmtrEventRecorderPrivate *priv =
+    emtr_event_recorder_get_instance_private (self);
+  g_autoptr(GVariant) event_id_variant = NULL;
+
+  g_return_val_if_fail (EMTR_IS_EVENT_RECORDER (self), NULL);
+  g_return_val_if_fail (_IS_VARIANT (aggregate_key), NULL);
+  g_return_val_if_fail (_IS_VARIANT (auxiliary_payload), NULL);
+
+  if (contains_maybe_variant (auxiliary_payload))
+    return NULL;
+
+  if (!priv->recording_enabled)
+    return NULL;
+
+  uuid_t parsed_event_id;
+  if (!parse_event_id (event_id, parsed_event_id))
+    return NULL;
+
+  event_id_variant = combine_event_id_with_key (parsed_event_id, NULL);
```

Comment:
I don't think this can be right. `emtr_aggregate_timer_new` passes its first argument (the return value of this function) to `emer_event_recorder_server_call_start_aggregate_timer` as the `event_id` parameter, which has GVariant type `ay`. But combine_event_id_with_key returns a variant of type `(aymv)`.

Reply:
Fixed